crates/nu-std/lib/help.nu

```diff
@@ -466,228 +746,59 @@ def show-operator [operator: record] {
-            if not ($flag.syntax_shape | is-empty) {
-                print -n $" <(ansi light_blue)($flag.syntax_shape)(ansi reset)>"
-            }
-            print $" - ($flag.description)"
-        }
-
-        print ""
-        print-help-header "Signatures"
-        for signature in $signatures {
-           let input = ($signature | where parameter_type == input | get 0)
-           let output = ($signature | where parameter_type == output | get 0)
-
-           print -n $"  <($input.syntax_shape)> | ($command.name)"
-           for positional in $positionals {
-               print -n $" <($positional.syntax_shape)>"
-           }
-           print $" -> <($output.syntax_shape)>"
-        }
-
-        if (not ($positionals | is-empty)) or $is_rest {
-            print ""
-            print-help-header "Parameters"
-            for positional in $positionals {
-                print -n "  "
-                if ($positional.is_optional) {
-                    print -n "(optional) "
-                }
-                print $"(ansi teal)($positional.parameter_name)(ansi reset) <(ansi light_blue)($positional.syntax_shape)(ansi reset)>: ($positional.description)"
-            }
+    mut target = ($item | str join " ")
 
-            if $is_rest {
-                let rest = ($parameters | where parameter_type == rest | get 0)
-                print $"  ...(ansi teal)rest(ansi reset) <(ansi light_blue)($rest.syntax_shape)(ansi reset)>: ($rest.description)"
-            }
-        }
+    if (not $find) and $target != "" {
+        $target = $"^($target)$"
     }
 
-    if not ($command.examples | is-empty) {
-        print ""
-        print-help-header -n "Examples"
-        for example in $command.examples {
-            print ""
-            print $"  ($example.description)"
-            print $"  > ($example.example | nu-highlight)"
-            if not ($example.result | is-empty) {
-                for line in (
-                    $example.result | table | if ($example.result | describe) == "binary" { str join } else { lines }
-                ) {
-                    print $"  ($line)"
-                }
-            }
-        }
+    let found_items = ($operators | find --regex $target)
+
+    match ($found_items | length) {
+        0 => {
+                operator-not-found-error (metadata $item).span
+        }, 
+        1 => {
+               show-operator $found_items.0
+                "" # return string
+        },
+        _ => {
+                $found_items
+        }        
     }
-
-    print ""
 }
 
-# Show help on nushell commands.
+# Show help on nushell commands. 
 export def "help commands" [
-    ...command: string@"nu-complete list-commands"  # the name of command to get help on
-    --find (-f): string  # string to find in command names and usage
+    ...item: string@"nu-complete list-commands"  # the name of command to get help on
+    --find (-f)     # show list of matches
```

Comment:
Question: Could the `find` pattern used here be based on aliases?

Reply:
Do you mean, could I specify an alias here and get help for the command name in the expansion of the alias?  If so, no.  
I just noticed that built-in help *does* do this.  I actually don't think it should, it's hiding too much information.  User typed in an alias, help should tell him/her about that alias.

Reply:
Yeah, that's what I meant. I don't have strong feelings about whether it should or should not be supported, but at least it should match what built-in help does.